src/main/java/org/opensearch/geospatial/index/common/xyshape/XYShapeConverter.java

```diff
@@ -102,7 +97,7 @@ public static XYCircle toXYCircle(Circle circle) {
      * @param point of type {@link Point}
      * @return {@link XYPoint} instance
      */
-    public static XYPoint toXYPoint(Point point) {
+    public static XYPoint toXYPoint(@NonNull Point point) {
```

Comment:
for all @NonNull annotation sending the NullPointerException is bad. I would recommend we should throw IllegalArgumentException. For lombok you can configure this via lombok.config file.

```
lombok.nonNull.exceptionType = IllegalArgumentException
```

Reply:
Here the value is null, hence it is thrown NPE. Why do you think it is bad? IMO, it will be very intuitive for null object to throw NPE . User will easily know what went wrong if the exception and reason is very clear.

Reply:
The idea of sending a NPE means the code was trying to accesses data from a Null object, but here how @NonNull annotation is interpreted is like this:
```
if(point == null) {
   throw NullPointerException("parameter is null");
}
```
which doesn't seem fair as we validating the object to be non null. But as the object was null so it make sense to have IllegalArgumentException(any exception which make sense here).

This is little subjective, but as per developer mind set seeing a NPE is bad and people look for cases where object was null and code is trying to access a field from that object which is not this use case.

Reply:
I would think it is fair, because we are interested in getX()/getY() and we are validating for that, since we want to fail sooner than later. IllegalArgumentException emphasis more on object value than object itself. if input is found null, user can immediately jump to place where it is created or instantiated to easily see what went wrong.

Reply:
"user can immediately jump to place where it is created or instantiated to easily see what went wrong"

This thing will be difficult as the stacktrace will come with the line which user can see only in the delombok code not in the normal code.

Nevertheless, this is quite a subjective topic. Please see if we still feel it should be a NPE instead of IllegalArgumentException